plugin/src/main/java/io/jenkins/plugins/forensics/git/miner/GitRepositoryMiner.java

```diff
@@ -49,11 +49,17 @@
     private static final long serialVersionUID = 1157958118716013983L;
 
     private final GitClient gitClient;
+    private final String latestCommitId;
 
-    GitRepositoryMiner(final GitClient gitClient) {
+    GitRepositoryMiner(final GitClient gitClient){
+        this(gitClient, null);
+    }
+
+    GitRepositoryMiner(final GitClient gitClient, final String latestCommitId) {
         super();
 
         this.gitClient = gitClient;
+        this.latestCommitId = latestCommitId;
```

Comment:
I think we need a second method here (and in the base class): 
```
    @Override
    public RepositoryStatistics mine(final RepositoryStatistics previousStatistics, final FilteredLog logger)
            throws InterruptedException {

```

and in the base class: 
```
@Override
    public RepositoryStatistics mine(final Collection<String> absoluteFileNames, final FilteredLog logger)
            throws InterruptedException {
         return mine(new RepositoryStatistics(), logger));
    }
```

then we can extract everything useful from the previous statistics. And for the first commit, the statistics are empty.